game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1404,35 +1404,33 @@ private void fireSuicideUnitsAttack() {
     if (suicideAttackers.isEmpty() || attackedDefenders.isEmpty()) {
       return;
     }
-    final boolean canReturnFire = (!isSuicideAndMunitionCasualtiesRestricted());
-    final List<Unit> allEnemyUnitsAliveOrWaitingToDie = new ArrayList<>();
-    allEnemyUnitsAliveOrWaitingToDie.addAll(defendingUnits);
+    final boolean canReturnFire = (!Properties.getSuicideAndMunitionCasualtiesRestricted(gameData));
```

Comment:
Should `canReturnFire` be a method of `Properties`? Would that be more direct and re-usable? Any concern that we might duplicate or confuse this check somewhere else?

Reply:
Having a method `Properties.canReturnFire(gameData)` seems like it would much simpler to understand here and would be much better encapsulation. Currently `MustFightBattle` is bloated and knows about far too much, it should not necessarily be in change of interpretting game rules like this.

Reply:
Noticing now as well we have the logic repeated: https://github.com/triplea-game/triplea/pull/5871/files/7e244f4d9c1c60f021bacdaf21c2606c914d4dbd#diff-009c223dfe7c716d07dc90f27899c933R1446, seems like an actual DRY violation where the idea of 'canReturnFire' is computed in multiple places, moving this logic would help with that problem.

Reply:
Uh, I think this was all removed. Are you looking at the latest diff and commits?

Reply:
Not sure, seems to indicate I need to start over on the entire review  :disappointed: